src/infrastructure/app/docs/ApiDocGenerator.ts

```diff
@@ -119,9 +119,12 @@ export class ApiDocGenerator implements IApiDocGenerator {
     this.apiDoc.info.license = info.license;
   }
 
-  private saveApiDoc(): void {
-    const filePath = resolve(join(__dirname, "../../../../openapi.json"));
-    writeFileSync(filePath, JSON.stringify(this.apiDoc, null, 2), "utf8");
+  saveApiDoc(): void {
+    const isGenerateJson = !(this.env !== DEV || !Object.keys(this.apiDoc.paths).length)

```

Comment:
Maybe `wasDocGenerated` is better and think about this way: 
```ts
  saveApiDoc(): void {
    const wasDocGenerated = !(this.env !== DEV || !Object.keys(this.apiDoc.paths).length);
    if (!wasDocGenerated) return;

    const filePath = resolve(join(__dirname, "../../../../openapi.json"));
    writeFileSync(filePath, JSON.stringify(this.apiDoc, null, 2), "utf8");

    SchemasStore.dispose();
  }
```

Reply:
done